[SRVCOM-2667] Improve the Serverless Overview page (#81607)
openshift/openshift-docs: about/about-serverless.adoc

====

// Add something about CLI tools
[id="about-serverless-components_{context}"]
== Components of {ServerlessProductName} 

[id="about-serverless-knative-serving_{context}"]
=== Knative Serving

Knative Serving builds on Kubernetes to support deploying and serving of applications and functions as serverless containers. Serving simplifies the application deployment, dynamically scales based on in incoming traffic and supports custom rollout strategies with traffic splitting. 

> Other features of Knative Serving include:

Reviewer: The important part from the IBM Style guide on lead-in sentences is to make sure the lead-in is a complete sentence.
```suggestion
Knative Serving includes the following features:
```

* Simplified deployment of serverless containers
* Traffic-based auto-scaling, including scale-to-zero
* Routing and network programming
* Point-in-time application snapshots and their configurations

[id="about-serverless-knative-eventing_{context}"]
=== Knative Eventing

Knative Eventing provides a platform that offers composable primitives to enable late-binding event sources and event consumers. 

Knative Eventing supports the following architectural cloud-native concepts:

* Services are loosely coupled during development and deployed independently to production.
* A producer can generate events before a consumer is listening, and a consumer can express an interest in an event or class of events that are not yet being produced.
* Services can be connected to create new applications without modifying producer or consumer, and with the ability to select a specific subset of events from a particular producer.

[id="about-serverless-functions_{context}"]
=== {FunctionsProductName}

{FunctionsProductName} allows you to write functions that are deployed as Knative Services and take advantage of Knative Serving and Eventing. 

> Other features of {FunctionsProductName} include:

Reviewer: See the comment above about using complete sentences to introduce a list.

=== {FunctionsProductName}
{FunctionsProductName} allows you to write functions that are deployed as Knative Services and take advantage of Knative Serving and Eventing. {FunctionsProductName} bring greater efficiency, more scalability and faster development to facilitate rapid go-to-market. Other features include:

* Build strategies including Source-to-Image (S2I) and Buildpacks
* Multiple runtimes, including Node.js, Quarkus and Go
* Local developer experience through the kn CLI
* Project templates
* Support for receiving CloudEvents and plain HTTP requests

=== {ServerlessLogicProductName}
{ServerlessLogicProductName} allows you to define declarative workflow models (YAML, JSON) that orchestrate event-driven, serverless applications and lets you visualize the execution of workflows, which simplifies debugging and optimization. It also includes built-in error handling and fault tolerance, making it easier to handle errors and exceptions that occur during the execution of a workflow. 
{ServerlessLogicProductName} implements the link:https://serverlessworkflow.io/[CNCF Serverless Workflow specification].

=== Knative CLI

> The Knative client `kn` allows you to create Knative resources from the command line or from within Shell scripts. With its extensive help pages and autocompletion support, it frees you from memorizing the detailed structure of the Knative resource schemas. `kn` features include:

Reviewer: ```suggestion
The Knative (`kn`) CLI allows you to create Knative resources from the command line or from within Shell scripts.  The Knative (`kn`) CLI includes the following features:
```
Minimalism suggestion for fluff. Formatting suggestions based on the [Knative CLI docs](https://docs.openshift.com/serverless/1.33/cli_tools/kn-plugins.html).

> * Full support for managing all features of Knative Serving: Services, Revisions and Routes

Reviewer: ```suggestion
* Support for managing the following Knative Serving features: 
** Services
** Revisions
** Routes
```

> * Support for managing Knative Eventing entities: Sources, Brokers, Triggers, Channels and Subscriptions

Reviewer: This would scan better and have a more parallel structure if you used subbullets to create a nested list.

See [Nested unordered lists](https://docs.asciidoctor.org/asciidoc/latest/lists/unordered/#nested-unordered-list) (Asciidoc docs)

> * A kubectl-like plugin architecture to extend the built-in functionality

Reviewer: ```suggestion
* A plugin architecture based on the design of the Kubernetes (`kubectl`) CLI tool
```

> * Easy integration of Knative into Tekton pipelines by using kn in a Tekton task

Reviewer: ```suggestion
* Integration of Knative into Tekton pipelines
```
See the ISG a[bout the term _easy_](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#easy). 
Minimalism: the last clause isn't adding much. If the part about Tetkon tasks is more important, consider leading with that part.
